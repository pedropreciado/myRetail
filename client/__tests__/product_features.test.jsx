import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProductFeatures from '../components/product/show/product_features';

configure({ adapter: new Adapter() });

let features = ['it works', 'it turns purple'];

describe('ProductFeatures', () => {
  let productFeatures = shallow(<ProductFeatures features={features}/>);

  it('renders correctly', () => {
    expect(productFeatures).toMatchSnapshot();
  });

  it('displays its title', () => {
    let title = productFeatures.find('#product-feature-container');

    expect(title.props().children[0].props.children).toBe('product highlights');
  });

  it('renders features', () => {
    let features = productFeatures.find('.feature-item');

    expect(features.length).toBe(2);
  });
});