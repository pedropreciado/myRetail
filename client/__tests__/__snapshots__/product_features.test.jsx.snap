// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFeatures renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductFeatures
    features={
        Array [
            "it works",
            "it turns purple",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          product highlights
</div>,
        <ul>
          <li
                    className="feature-item"
          >
                    it works
          </li>
          <li
                    className="feature-item"
          >
                    it turns purple
          </li>
</ul>,
      ],
      "id": "product-feature-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "product highlights",
        },
        "ref": null,
        "rendered": "product highlights",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="feature-item"
>
              it works
</li>,
            <li
              className="feature-item"
>
              it turns purple
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "it works",
              "className": "feature-item",
            },
            "ref": null,
            "rendered": "it works",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "it turns purple",
              "className": "feature-item",
            },
            "ref": null,
            "rendered": "it turns purple",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            product highlights
</div>,
          <ul>
            <li
                        className="feature-item"
            >
                        it works
            </li>
            <li
                        className="feature-item"
            >
                        it turns purple
            </li>
</ul>,
        ],
        "id": "product-feature-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "product highlights",
          },
          "ref": null,
          "rendered": "product highlights",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="feature-item"
>
                it works
</li>,
              <li
                className="feature-item"
>
                it turns purple
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "it works",
                "className": "feature-item",
              },
              "ref": null,
              "rendered": "it works",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "it turns purple",
                "className": "feature-item",
              },
              "ref": null,
              "rendered": "it turns purple",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
